src/curl.c

```diff
@@ -131,14 +131,23 @@ int GetAccessTokenFromIMDS(const char *type, MemoryStruct *accessToken)
   {
     Log(LogLevel_Error, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
     free(accessToken->memory);
+    accessToken->memory = 0;
```

Comment:
I added this because if you don't see memory to 0, then https://github.com/microsoft/AzureKeyVaultManagedHSMEngine/blob/main/src/dllmain.c#L246-L247 will cause a double free on accessToken.